Vagrantfile

```diff
@@ -65,7 +65,13 @@ Vagrant.configure(2) do |config|
 
     # Change grub to support cgroups memory if necessary
     config.vm.provision "shell", path: "cookbook/system-config/grub-enable-cgroup-memory.sh"
-    config.vm.provision :reload
+    fileCheck = lambda {
+        if File.exists?("VAGRANT_NEEDS_RELOAD")
+            File.delete("VAGRANT_NEEDS_RELOAD")
+            return true
+        end
```

Comment:
Wouldn't it be nicer if you returned false instead of nil so it would return a boolean?